test/taskrun_test.go

```diff
@@ -0,0 +1,146 @@
+// +build e2e
+
+/*
+Copyright 2018 Knative Authors LLC
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"testing"
+
+	buildv1alpha1 "github.com/knative/build/pkg/apis/build/v1alpha1"
+	knativetest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/knative/build-pipeline/pkg/apis/pipeline/v1alpha1"
+
+	// Mysteriously by k8s libs, or they fail to create `KubeClient`s from config. Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+const (
+	hwTaskName    = "helloworld"
+	hwTaskRunName = "helloworld-run"
+)
+
+func getHelloWorldTask(namespace string) *v1alpha1.Task {
+	return &v1alpha1.Task{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      hwTaskName,
+		},
+		Spec: v1alpha1.TaskSpec{
+			BuildSpec: &buildv1alpha1.BuildSpec{
+				Steps: []corev1.Container{
+					corev1.Container{
+						Name:  "helloworld-busybox",
+						Image: "busybox",
+						Args: []string{
+							"echo", "hello world",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getHelloWorldTaskRun(namespace string) *v1alpha1.TaskRun {
+	return &v1alpha1.TaskRun{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      hwTaskRunName,
+		},
+		Spec: &v1alpha1.TaskRunSpec{
+			TaskRef: v1alpha1.TaskRef{
+				Name: hwTaskName,
+			},
+			Trigger: v1alpha1.TaskTrigger{
+				TriggerRef: v1alpha1.TaskTriggerRef{
+					Type: v1alpha1.TaskTriggerTypeManual,
+				},
+			},
+		},
+	}
+}
+
+// TestTaskRun is an integration test that will verify a very simple "hello world" TaskRun can be
+// executed.
+func TestTaskRun(t *testing.T) {
+	t.Skip("Will fail until #59 is completed :D")
+	// TODO: when committing, actually skip this test since it won't pass yet
+
+	logger := logging.GetContextLogger(t.Name())
+	c, namespace := setup(t, logger)
+
+	knativetest.CleanupOnInterrupt(func() { tearDown(logger, c.KubeClient, namespace) }, logger)
+	defer tearDown(logger, c.KubeClient, namespace)
+
+	// Create task
+	_, err := c.TaskClient.Create(getHelloWorldTask(namespace))
```

Comment:
In the build repo I'm a total pain about preferring the form where it's clearer that `err` doesn't escape this scope so you don't have to think about whether it's referenced later:

```
if _, err := doStuff(); err != nil {
  // handle
}
// err isn't defined out here
```

I find it's helpful when reading to be able to mentally drop `err` from the list of things I care about.

When Go2 has better error handling these'll all just become [`check` calls](https://go.googlesource.com/proposal/+/master/design/go2draft-error-handling.md).

Reply:
kk ill try to use this when i can! thanks for the pointer. @tejal29 was pointing this out to me as well when we were pairing but i keep forgetting, plz let me know if you see me miss it again!

(The only thing I don't like is when this syntax sometimes forces you to declare `var err error`)